ibis/util.py

```diff
@@ -355,3 +356,39 @@ def flatten_iterable(iterable):
             yield from flatten_iterable(item)
         else:
             yield item
+
+
+def warn_deprecated(name, ver='', instead='', ignoreframes=1):
+    """Warn about deprecated usage, including stacktrace, and
+    what to do instead."""
+    warnlines = []
+    msg = f'"{name}" is deprecated'
+    if ver:
+        msg += f' since v{ver}'
+
+    if instead:
+        msg += f'; use {instead}.'
+
+    warnlines.append(msg)
+
+    import traceback
+
+    warnlines.append('Deprecated call traceback (most recent last):')
+    for f in traceback.extract_stack(limit=6)[:-ignoreframes]:
+        warnlines.append(f'{f.filename} at line {f.lineno} in {f.name}')
+    warnings.warn('\n'.join(warnlines), DeprecationWarning)
```

Comment:
`DeprecationWarning` isn't shown by default (https://docs.python.org/3/library/exceptions.html#DeprecationWarning), unless you're in the `__main__` module, which isn't the primary way ibis is used. We have to use something else instead, and I think `FutureWarning` is probably the okay-est type for that :)